### DISCUSSION TITLE
LSP Configuration

### DISCUSSION THREAD

**Discussion by javaldrnld** [LazyVim/LazyVim#723] [Q&A] [answered]
Where should the nvim LSP configuration file be placed in order to properly configure it, as shown in the picture? So that I can add the clangd, pyright lsp config 
![image](https://user-images.githubusercontent.com/103297695/236689083-c05aeb3e-4350-47b6-afdd-b96b8e9889cc.png)

**Comment by Jomik** [comment, marked as answer]
You can look in the docs, there is an example [here](https://lazyvim.org/plugins/extras/lang.json#nvim-lspconfig) for `jsonls`.
All you should need is a plugin that has the below, to install `pyright`, `solargraph` and `clangd`.
LazyVim will automatically pass in `capabilities`. Look in the docs if you need to pass in additional settings to the language server 👍 
```lua
return {
  "neovim/nvim-lspconfig",
  opts = {
    -- make sure mason installs the server
    servers = {
      pyright = {},
      solargraph = {},
      clangd = {},
    },
  },
}
```

**Reply by javaldrnld** [reply]
All of the LSP config that I will use will be put on the server and it is not necessary to do the `require("language").setup{}`?

**Reply by javaldrnld** [reply]
When I do the snippet you gave, and add the ruff and black it is the result. Or is it because black and ruff is not lsp that's why?
![image](https://user-images.githubusercontent.com/103297695/236727777-3f310fa1-3150-4a46-a830-74ebc24b8f3f.png)

**Reply by Jomik** [reply]
I do not see any `black`, but `ruff` should probably be `ruff_lsp`.
You should look into the message it gives you. It comes from `lspconfig`, as it says in the beginning. So we can go to that repo and find the [mentioned file](https://github.com/neovim/nvim-lspconfig/blob/master/doc/server_configurations.md)

**Reply by javaldrnld** [reply]
I solved the problem, thanks!!
